pkg/controller/core/workload_controller_test.go

```diff
@@ -2004,13 +2106,71 @@ func TestReconcile(t *testing.T) {
 			},
 			wantError: nil,
 		},
+		"reconcile DRA ResourceClaim but no quota and WL suspended": {
+			workload: func() *kueue.Workload {
+				wl := utiltesting.MakeWorkload("wl-no-quota-suspended", "ns").
+					Queue("lq").
+					// Reserve once to simulate prior reservation, then mark admitted+suspended
+					ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("gpus", "flavor1", "1").Obj()).
+					Admitted(true).
+					Active(false).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Obj()).
+					Obj()
+				// DRA: reference claim rc1
+				wl.Spec.PodSets[0].Template.Spec.ResourceClaims = []corev1.PodResourceClaim{{
+					Name: "gpu", ResourceClaimName: ptr.To("rc1"),
+				}}
+				wl.Spec.PodSets[0].Template.Spec.Containers = []corev1.Container{{
+					Name: "main", Image: "pause",
+					Resources: corev1.ResourceRequirements{Claims: []corev1.ResourceClaim{{Name: "gpu"}}},
+				}}
+				return wl
+			}(),
+			// ClusterQueue with insufficient quota: flavor has 0 for gpus
+			cq: utiltesting.MakeClusterQueue("cq").
+				ResourceGroup(
+					*utiltesting.MakeFlavorQuotas("flavor1").
+						Resource("gpus", "0").Obj(),
+				).Obj(),
+			lq: utiltesting.MakeLocalQueue("lq", "ns").ClusterQueue("cq").Obj(),
+			wantEvents: []utiltesting.EventRecord{{
+				Key:       types.NamespacedName{Namespace: "ns", Name: "wl-no-quota-suspended"},
+				EventType: corev1.EventTypeNormal,
+				Reason:    "EvictedDueToDeactivated",
+				Message:   "The workload is deactivated",
+			}},
+		},
 	}
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
 			features.SetFeatureGateDuringTest(t, features.ObjectRetentionPolicies, tc.enableObjectRetentionPolicies)
 
 			objs := []client.Object{tc.workload}
 			clientBuilder := utiltesting.NewClientBuilder().WithObjects(objs...).WithStatusSubresource(objs...).WithInterceptorFuncs(interceptor.Funcs{SubResourcePatch: utiltesting.TreatSSAAsStrategicMerge})
+			// Inject DRA ResourceClaim when present (for the DRA test case)
+			if tc.workload != nil && tc.workload.Namespace == "ns" &&
+				(tc.workload.Name == "wlWithDRAResourceClaim" || tc.workload.Name == "wl-no-quota-suspended") {
+				// Create v1beta2 ResourceClaim named rc1 in ns
+				count := int64(1)
+				if tc.workload.Name == "wl-no-quota-suspended" {
+					count = 3 // Request 3 resources but quota is only 2
+				}
```

Comment:
Could you please instead extend the `cases` struct with the list of ResourceClaims and ResourceClaimsTemplates, so that we can declaratively specify the input for the tests.

Reply:
refactored